exonum/src/api/public/system.rs

```diff
@@ -25,18 +25,40 @@ struct MemPoolInfo {
     pub size: usize,
 }
 
-#[doc(hidden)]
+/// Contains amount of connected peers
 #[derive(Deserialize, Serialize, Clone, Debug, PartialEq)]
-pub struct HealthCheckInfo {
-    pub connectivity: bool,
+pub struct PeersAmount {
+    /// amount of connected peers
+    pub amount: usize,
 }
 
-/// ConsensusStatusInfo shows the possibility to achieve the consensus between validators
-/// in current state.
+/// Shows connectivity status of the node
 #[derive(Deserialize, Serialize, Clone, Debug, PartialEq)]
-pub struct ConsensusStatusInfo {
-    /// Consensus status: true - if consensus achieved, else - false.
-    pub status: bool,
+pub enum ConnectivityStatus {
+    /// The node has no peers
+    NotConnected,
+    /// The node has amount of peers
+    Connected(PeersAmount),
+}
+
+/// Shows status of the consensus
+#[derive(Deserialize, Serialize, Clone, Debug, PartialEq)]
+pub enum ConsensusStatus {
+    /// Consensus disabled
+    Disabled,
+    /// Consensus enabled but not active
+    Enabled,
+    /// Consensus enabled and active
+    Active,
+}
+
+/// Information about connectivity and consensus
```

Comment:
Documentation for this structure and its fields doesn't seem being helpful at all.